### ISSUE TITLE
Unclear documentation

### ISSUE THREAD

**Issue by surdu** [jrullan/StateMachine#15]
Hello,

Just started using your library and this puzzles me:

```
S0->addTransition(&transitionS0,S0);    // Transition to itself (see transition logic for details)
```

Where exactly is the "transition logic" explained ?

**Comment by olehvavryniv** [comment]
Good point. I figured out that the `transitionS0` function should always return false. In other cases, you will loop in S0 forever.
Also, from the source code, I see that transition to itself for the first state is not required.
